src/main/java/io/cdap/e2e/pages/actions/CdfPipelineRunAction.java

```diff
@@ -17,67 +17,129 @@
 package io.cdap.e2e.pages.actions;
 
 import io.cdap.e2e.pages.locators.CdfPipelineRunLocators;
+import io.cdap.e2e.utils.AssertionHelper;
 import io.cdap.e2e.utils.ConstantsUtil;
 import io.cdap.e2e.utils.ElementHelper;
+import io.cdap.e2e.utils.PageHelper;
 import io.cdap.e2e.utils.SeleniumDriver;
 import io.cdap.e2e.utils.SeleniumHelper;
-import io.cdap.e2e.utils.WaitHelper;
 import org.junit.Assert;
-import org.openqa.selenium.NoSuchElementException;
 import org.openqa.selenium.StaleElementReferenceException;
 import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.ArrayList;
-
 /**
- * Represents CdfPipelineRunAction
+ * Represents Cdf Pipeline Run Page Actions
  */
 public class CdfPipelineRunAction {
   private static final Logger logger = LoggerFactory.getLogger(CdfPipelineRunAction.class);
-  public static CdfPipelineRunLocators cdfPipelineRunLocators;
 
   static {
-    cdfPipelineRunLocators = SeleniumHelper.getPropertiesLocators(CdfPipelineRunLocators.class);
+    SeleniumHelper.getPropertiesLocators(CdfPipelineRunLocators.class);
+  }
+
+  /**
+   * Click on the Run button
+   */
+  public static void runClick() {
+    ElementHelper.clickOnElement(CdfPipelineRunLocators.run);
   }
 
-  public static void runClick() throws InterruptedException {
-    ElementHelper.clickOnElement(cdfPipelineRunLocators.run);
+  /**
+   * Click on the Run button inside 'Runtime Arguments' dialog
+   */
+  public static void clickDeployedConfigRunButton() {
+    ElementHelper.clickOnElement(CdfPipelineRunLocators.deployedConfigRunButton);
   }
 
+  /**
+   * Get the Pipeline's Status bubble color
+   *
+   * @return color attribute value
+   */
   public static String runPipelineStatus() {
-    return cdfPipelineRunLocators.runPipelineStatus.getAttribute(ConstantsUtil.COLOR);
+    return CdfPipelineRunLocators.runPipelineStatus.getAttribute(ConstantsUtil.COLOR);
   }
 
-  public static Boolean isRunning() {
-    return provideState(cdfPipelineRunLocators.runningStatus);
+  /**
+   * Check if the Pipeline is in the expected Status
+   *
+   * @param element WebElement of Pipeline's expected Status
+   * @return boolean
+   */
+  private static boolean provideState(WebElement element) {
```

Comment:
can we rename this method to `validatePipelineExpectedStatus` ?

Reply:
Updated.